app/questionnaire/navigator.py

```diff
@@ -266,7 +238,7 @@ def get_latest_location(self, completed_blocks=None):
         """
         location_path = self.get_location_path()
         if completed_blocks:
-            incomplete_blocks = [item for item in location_path if item not in completed_blocks]
+            incomplete_blocks = [item for item in location_path if item.__dict__ not in completed_blocks]
```

Comment:
You could allow `Location.__eq__` to check if `other` is a dict and do a dict comparison so you can do `Location == {...}` ?